cmd/eval.go

```diff
@@ -111,6 +114,13 @@ func validateEvalParams(p *evalCommandParams, cmdArgs []string) error {
 	} else if !p.partial && of == evalSourceOutput {
 		return errors.New("invalid output format for evaluation")
 	}
+
+	if p.optimizationLevel > 0 {
+		if len(p.dataPaths.v) > 0 && p.bundlePaths.isFlagSet() {
+			return fmt.Errorf("specify either --data or --bundle flag with optimization level greater than 0")
+		}
+	}
+
```

Comment:
If we allow users to set both the `--data` and `--bundle` flag with the `optimize` flag > 0, we'd have to merge all the paths before we build an optimized bundle. But it would lead to complications like managing roots, handling merges between bundle and non-bundle paths etc. 

Other option if we allow both the `--data` and `--bundle` flag with the `optimize` flag > 0, is to build two optimized bundles from the `--data` and `--bundle` sources and then merging the bundles for eval. Here too we'll have to deal with questions around individual  bundle roots while merging them.